src/tools/components.cpp

```diff
@@ -111,7 +107,7 @@ int main(int argc, char *argv[])
     }
 
     std::vector<osrm::tools::TarjanEdge> graph_edge_list;
-    auto number_of_nodes = osrm::tools::loadGraph(argv[1], coordinate_list, graph_edge_list);
+    auto number_of_nodes = osrm::tools::loadGraph(std::string(argv[1]), coordinate_list, graph_edge_list);
```

Comment:
`const char*` should converted to `std::string` automatically (implicit conversion). You can remove the `std::string()` I think.